app/components/goodcity/code-search-overlay.js

```diff
@@ -0,0 +1,130 @@
+import Ember from "ember";
+import _ from "lodash";
+import SearchMixin from "stock/mixins/search_resource";
+
+/**
+ * An overlay that pops up from the bottom of the screen, allowing the user
+ * to search and select an order.
+ *
+ * The popup *does not* do anythinng to the order apart from returning it
+ *
+ * @property {boolean} open whether the popup is visible or not
+ * @property {function} onSelect callback triggered when an order is selected
+ */
+export default Ember.Component.extend(SearchMixin, {
+  fetchMoreResult: true,
+  store: Ember.inject.service(),
+  filter: "",
+  searchText: "",
+  fetchMoreResult: true,
+  storageType: null,
+  isSearchCodePreviousRoute: Ember.computed.localStorage(),
+
+  init() {
+    this._super("code-search-overlay");
+  },
+
+  allPackageTypes: Ember.computed("open", function() {
+    return this.get("store").peekAll("code");
+  }),
+
+  filteredPackageTypes: Ember.computed("allPackageTypes", function() {
+    const pkgTypes = this.get("allPackageTypes");
+    const storageType = this.get("storageType");
+    if (["Box", "Pallet"].indexOf(storageType) > -1) {
+      return pkgTypes.filterBy(`allow_${storageType.toLowerCase()}`, true);
+    }
+    return pkgTypes;
+  }),
+
+  hasSearchText: Ember.computed("searchText", function() {
+    return Ember.$.trim(this.get("searchText")).length;
+  }),
+
+  hasFilter: Ember.computed("filter", function() {
+    return Ember.$.trim(this.get("filter")).length;
+  }),
+
+  onSearchTextChange: Ember.observer("searchText", function() {
+    // wait before applying the filter
+    Ember.run.debounce(this, this.applyFilter, 500);
+  }),
+
+  applyFilter: function() {
+    this.set("filter", this.get("searchText"));
+    this.set("fetchMoreResult", true);
+  },
+
+  clearHiglight() {
+    Ember.$("em").replaceWith(function() {
+      return this.innerHTML;
+    });
+  },
+
+  highlight() {
```

Comment:
If these highlighting logic repeated, find a way to make it more generic. Try to avoid using of `$` here.